split-gpg2-client

```diff
@@ -1,21 +1,35 @@
-#!/bin/bash
+#!/bin/bash --
+set -euo pipefail
 
 for d in /etc "${XDG_CONFIG_HOME:-$HOME/.config}"; do
     rc_file="$d/split-gpg2-rc"
-    if [ -r "$rc_file" ]; then
+    if [[ -r "$rc_file" ]]; then
         . "$rc_file"
     fi
 done
 
-if [[ -z "$SPLIT_GPG2_SERVER_DOMAIN" ]]; then
+if ! [[ -v SPLIT_GPG2_SERVER_DOMAIN ]]; then
     SPLIT_GPG2_SERVER_DOMAIN="@default"
+elif [[ "$SPLIT_GPG2_SERVER_DOMAIN" =~ ^[A-Za-z@][0-9A-Za-z_-]*$ ]]; then
+    :
+else
+    # prevent accidental command injection into socat
+    printf '$SPLIT_GPG2_SERVER_DOMAIN (%q) is not a valid qrexec target\n' "$SPLIT_GPG2_SERVER_DOMAIN" >&2
+    exit 1
 fi
 
-agent_socket="$(gpgconf --list-dirs | grep '^agent-socket:' | cut -d ':' -f 2)"
-if [[ -z "$agent_socket" ]]; then
-    echo "Failed to find gpg-agent socket path" >&2
+agent_socket=$(gpgconf --list-dirs -o/dev/stdout |
+    LC_ALL=C awk -F: '/^agent-socket:/ { print $2 }')
```

Comment:
This seems to make practically no speed difference so I prefer the grep+cut since it's more obvious what it does.

Reply:
This is a matter of opinion: to me, awk is more obvious since I hardly ever use cut.